include/sas.h

```diff
@@ -304,6 +304,10 @@ class SAS
                                     const char *fmt,
                                     ...);
 
+  typedef int (socket_factory_t)(const char* hostname,
+                                 const char* port);
```

Comment:
What are the requirements of this function? I'm mainly thinking about what socket options must / must not / may be set on the socket.